sklearn/linear_model/_quantile.py

```diff
@@ -0,0 +1,225 @@
+# Authors: David Dale <dale.david@mail.ru>
+#          Christian Lorentzen <lorentzen.ch@gmail.com>
+# License: BSD 3 clause
+import numpy as np
+from scipy.optimize import linprog
+
+from ..base import BaseEstimator, RegressorMixin
+from ._base import LinearModel
+from ..utils.validation import _check_sample_weight
+from ..utils.fixes import sp_version, parse_version
+
+
+class QuantileRegressor(LinearModel, RegressorMixin, BaseEstimator):
+    """Linear regression model that predicts conditional quantiles.
+
+    The linear Quantile Regressor optimizes the pinball loss for a
+    desired `quantile` and is robust to outliers.
+
+    Read more in the :ref:`User Guide <quantile_regression>`
+
+    .. versionadded:: 1.0
+
+    Parameters
+    ----------
+    quantile : float, default=0.5
+        The quantile that the model tries to predicts. It must be strictly
+        between 0 and 1.
+
+    alpha : float, default=1.0
+        Constant that multiplies L1 penalty term.
+
+    fit_intercept : bool, default=True
+        Whether or not to fit the intercept.
+
+    solver : {'highs-ds', 'highs-ipm', 'highs', 'interior-point', \
+            'revised simplex'}, default='interior-point'
+        Method used by scipy.optimize.linprog to solve the linear programming
+        formulation. Note that 'highs' is recommended for usage with
+        scipy>=1.6.0.
+
+    solver_options : dict, default=None
+        Additional parameters passed to scipy.optimize.linprog as options.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features,)
+        Estimated coefficients for the features.
+
+    intercept_ : float
+        The intercept of the model, aka bias term.
+
+    n_iter_ : int
+        The actual number of iterations performed by the solver.
```

Comment:
I know we are not consistant here but I would add a See Also section to point to HuberRegressor and Lasso